pkg/network/usm/shared_libraries.go

```diff
@@ -328,24 +319,22 @@ func (r *soRegistry) register(root, libPath string, pid uint32, rule soRule) {
 		return
 	}
 
-	r.m.Lock()
-	defer r.m.Unlock()
-	if _, found := r.blocklistByID[pathID]; found {
+	if _, found := r.blocklistByID.Load(pathID); found {
```

Comment:
By transitioning from a scope-wise locking semantic to a function-wise locking semantic, we are altering the way locks are implemented. Are we confident that this change won't introduce any new race conditions? For example, considering that this function can be invoked multiple times, the state of the maps might change during the actual execution. Previously, the function would have been blocked by `r.m.Lock()`

Reply:
Yes.
That's why I used LoadOrStore multiple times around this function.
And if you read again that function you can see that only once we can register the hook